ambry-store/src/main/java/com.github.ambry.store/StoreMessageReadSet.java

```diff
@@ -171,7 +198,20 @@ public long writeTo(int index, WritableByteChannel channel, long relativeOffset,
     long startOffset = options.getOffset() + relativeOffset;
     long sizeToRead = Math.min(maxSize, options.getMessageInfo().getSize() - relativeOffset);
     logger.trace("Blob Message Read Set position {} count {}", startOffset, sizeToRead);
-    long written = options.getChannel().transferTo(startOffset, sizeToRead, channel);
+    long written = 0;
+    long bufStartOffset = relativeOffset - options.getPreFetchedDataRelativeOffset();
+    if (options.getPreFetchedDataRelativeOffset() != -1 && bufStartOffset < 0) {
+      logger.error("RelativeOffset out of boundary.", relativeOffset, options.getPreFetchedDataRelativeOffset());
+    }
+    if (options.getPreFetchedDataRelativeOffset() != -1 && bufStartOffset >= 0 && bufStartOffset <= Integer.MAX_VALUE
```

Comment:
minor: `options.getPreFetchedDataRelativeOffset() != -1 && bufStartOffset >= 0 && bufStartOffset + sizeToRead <= Integer.MAX_VALUE` should be enough, right?

Reply:
fixed.